web_full_stack/js/src/CategoriesPage.js

```diff
@@ -1,53 +1,48 @@
-import React, { Component, Fragment } from 'react';
+import React, { Fragment, useState, useEffect } from 'react';
 import { fetchCategories } from './API';
 
-export default class CategoriesPage extends Component {
-  state = {
-    categories: [],
-  }
-
-  async componentDidMount() {
-    const resp = await fetchCategories();
+export default function CategoriesPage() {
+  const [categories, setCategories] = useState([]);
+  const [query, setQuery] = useState();
 
-    this.setState({
-      categories: resp.categories,
+  useEffect(() => {
+    fetchCategories().then((response) => {
```

Comment:
What's the rationale for not using `async / await` here? Is this to present candidates with the opportunity to refactor and or demonstrate familiarity of older syntax?

Reply:
no - my idea is that the newer function syntax is easier to read top-down, and if you're starting from scratch on a React project, you're probably reaching for function components first instead of class components.

Reply:
No real reason, although it would require creating an additional inner function, but that's probably ok.